snowflake/datadog_checks/snowflake/config.py

```diff
@@ -57,12 +58,15 @@ def __init__(self, instance=None):
             if password is None:
                 raise ConfigurationError('Must specify a password if using snowflake authentication')
 
-        else:
-            if authenticator == 'oauth' and token is None:
-                raise ConfigurationError('If using OAuth, you must specify a token')
+        elif authenticator == 'oauth':
+            if token is None and token_path is None:
+                raise ConfigurationError('If using OAuth, you must specify a `token` or a `token_path`')
 
-            if authenticator not in self.AUTHENTICATION_MODES:
-                raise ConfigurationError('The Authenticator method set is invalid: {}'.format(authenticator))
+            if token and token_path:
+                raise ConfigurationError('`token` and `token_path` are set, please set only one option')
+
```

Comment:
Should we add validation for if `token_path` is not set, `password` should be set?

Reply:
I'll create a new PR to refactor validation after this PR and this one https://github.com/DataDog/integrations-core/pull/11180 are merged, because it's adding another way to connect